core/tee/tee_rpmb_fs.c

```diff
@@ -1192,6 +1197,10 @@ static TEE_Result tee_rpmb_read(uint16_t dev_id, uint32_t addr, uint8_t *data,
 	blk_idx = addr / RPMB_DATA_SIZE;
 	byte_offset = addr % RPMB_DATA_SIZE;
 
+	if (len + byte_offset + RPMB_DATA_SIZE < RPMB_DATA_SIZE) {
```

Comment:
Woudl prefer `if (ADD_OVERFLOW(len, byte_offset, &tmp)) return ...;`

Reply:
Then two `ADD_OVERFLOW()` calls would be needed, no? (we want to make sure the `ROUNDUP()` does not overflow).

Reply:
ok, agree. You're initial proposal is fine.